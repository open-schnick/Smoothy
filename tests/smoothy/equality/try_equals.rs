use crate::failing_assertion;
use smoothy::{assert_that, EqualityAssertion};

mod succeeds {
    use super::*;

    #[test]
    fn with_numbers() {
        assert_that(42u8).try_into_equals(42i8);
    }
}

mod fails {
    use super::*;

    #[test]
    fn with_conversion_error() {
        failing_assertion!({
            assert_that(42u8).try_into_equals(-42i8);
        });
    }

    #[test]
    #[should_panic = "assertion failed: `(left == right)`"]
    fn with_numbers() {
        assert_that(42u8).try_into_equals(100i8);
    }
}
