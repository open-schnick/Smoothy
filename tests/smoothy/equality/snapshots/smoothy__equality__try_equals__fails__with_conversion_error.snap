---
source: tests/smoothy/equality/try_equals.rs
expression: assertion_failed_output
---
assertion failed: `(TryInto conversion succeeds)`
           found:  Err(TryFromIntError(()))
